### DISCUSSION TITLE
use bear -- make to build compile databs

### DISCUSSION THREAD

**Discussion by chchc086** [rizsotto/Bear#589] [Q&A]
i used bear -- [my_compile_command] to build compile_commands.json
when my makefile written as below, **it works**
![image](https://github.com/user-attachments/assets/db824c35-4557-4dde-8259-dc25de33d912)
**but when i write it like this**, it not work, and the error information is " no such file or directory" ,i find ($PWD) is empty,
![image](https://github.com/user-attachments/assets/2752467e-a07c-4427-80b1-861b839cf38a)
but it works only build my code use  [my_compile_command] 
why bear recognizes $(PWD) as empty ?

**Comment by rizsotto** [comment]
That's an interesting one @chchc086 !

Bear does not read the makefile. It intercepts the commands the build process executes. And it looks like `make` behaves differently. But Bear executes `make` as a child process, and keeps the environment variables and the current working directory as is.
